### ISSUE TITLE
compact-xapian-database fails to restart alaveteli under systemd

### ISSUE THREAD

**Issue by sagepe** [mysociety/alaveteli#4288]
The `compact-xapian-database` script includes the following line:
```
commonlib/bin/output-on-error /usr/sbin/service "$DAEMON_NAME" restart
```
This fails on Debian Stretch (and probably any other systemd-based OS) when run as an app user as the `service` command just passes this to `systemctl` which needs root privileges.

For the purposes of this script, either running `/etc/init.d/${DAEMON_NAME} restart` directly (as the script does nothing that the app user can't do) or even simply touching the `alaveteli/tmp/restart.txt` file would probably be enough to resolve this.

Happy to submit a patch and PR, but thought we might want to discuss the preferred option and any other potential places this might bite.

**Comment by garethrees** [comment]
Haven't looked in to this yet, but just while its in my mind…

Our install docs specifically set the permissions for this:

```
chown root:alaveteli /etc/init.d/alaveteli
chmod 754 /etc/init.d/alaveteli
service alaveteli start
```

Via http://alaveteli.org/docs/installing/manual_install/#generate-application-daemon

**Comment by garethrees** [comment]
Also noting that we had to call `/etc/init.d/${DAEMON_NAME}` directly in https://github.com/mysociety/alaveteli/issues/2373

**Comment by sagepe** [comment]
Just to clarify, it's not the permissions on the script in `/etc/init.d` that's the issue, it's the use of the `service` command as a wrapper under systemd. Anywhere that currently calls `service` as the application user will need review.

Given a bit of time we could look at whether there is a more elegant systemd way of handling this kind of event.

**Comment by garethrees** [comment]
https://github.com/mysociety/alaveteli/blob/develop/script/compact-xapian-database#L23 is the only place that calls `service`

**Comment by garethrees** [comment]
> even simply touching the `alaveteli/tmp/restart.txt` file would probably be enough to resolve this.

This wouldn't work for non-passenger installs, which was why we unified the interface with the `/etc/init.d` script.

I think calling the script directly is the easiest way out of this situation for now.

**Comment by garethrees** [comment]
Made a PR with the proposed change https://github.com/mysociety/alaveteli/pull/4292

**Comment by garethrees** [comment]
Fixed by 663bb16df54870fcab0e15a6af36647991cdca9e
